### ISSUE TITLE
invalid byte sequence in US-ASCII

### ISSUE THREAD

**Issue by adrienkohlbecker** [brigade/haml-lint#94]
Hello,

I'm unable to lint files containing UTF-8 characters. Here is an example
```haml
%p Testing encodings àéù
```

Here is the stack trace:

```
invalid byte sequence in US-ASCII
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/document.rb:41:in `split'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/document.rb:41:in `process_source'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/document.rb:32:in `initialize'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/runner.rb:52:in `new'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/runner.rb:52:in `collect_lints'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/runner.rb:21:in `block in run'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/runner.rb:20:in `map'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/runner.rb:20:in `run'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/cli.rb:82:in `scan_for_lints'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/cli.rb:51:in `act_on_options'
/usr/local/bundle/gems/haml_lint-0.15.1/lib/haml_lint/cli.rb:23:in `run'
/usr/local/bundle/gems/haml_lint-0.15.1/bin/haml-lint:7:in `<top (required)>'
/usr/local/bundle/bin/haml-lint:23:in `load'
/usr/local/bundle/bin/haml-lint:23:in `<main>'
Report this bug at https://github.com/brigade/haml-lint/issues

To help fix this issue, please include:
- The above stack trace
- Haml-Lint version: 0.15.1
- Haml version: 4.0.7
- RuboCop version: 0.34.0
- Ruby version: 2.2.3
```

Best

**Comment by sds** [comment]
Hey @adrienkohlbecker, thanks for the report.

When I've seen issues like this reported in the past it's usually an editor encoding issue. Notice that the string is being interpreted as US-ASCII.

Having said that, I think `haml-lint` should display a better error in this case. Can I get you to run:

```
File.read('test-file.haml').bytes
```

...against a file that produces this problem on your system and post the output here so I can attempt to reproduce on my end? Thanks!

**Comment by adrienkohlbecker** [comment]
Sure, here you go:
```
irb(main):004:0> puts File.read("test.haml")
%p Test àéùö
=> nil
irb(main):005:0> puts File.read("test.haml").bytes
[37, 112, 32, 84, 101, 115, 116, 32, 195, 160, 195, 169, 195, 185, 195, 182, 10]=> nil
```

I've confirmed this file produces the above stack-trace.
If it's of any use, I'm using Atom as an editor (which reports utf-8), and this is run inside docker, with the `ruby:2.2.3` image

**Comment by sds** [comment]
Thanks, @adrienkohlbecker.

The likely explanation is that your environment might have the `LC_ALL=C` or `LC_ALL=en_US.US-ASCII`, or some other environment-specific quirk is causing Ruby to assume an incorrect default encoding of `US-ASCII`.

We will just assume all HAML files are encoded as UTF-8. This is a sane default for the vast majority of users and should avoid weird issues such as these.

See ad93903fff77030ce702d6995771864f8602cce2 and https://github.com/bbatsov/rubocop/issues/1289. The moment someone finds a need to lint HAML files encoded as something other than UTF-8, they are welcome to submit a pull request that allows the specified encoding to be forced.

**Comment by adrienkohlbecker** [comment]
Thanks! I've confirmed it's working now in my environment.
Any chance for a release on rubygems?

**Comment by sds** [comment]
Released [0.15.2](https://rubygems.org/gems/haml_lint/versions/0.15.2)

**Comment by adrienkohlbecker** [comment]
Nice, thanks :+1:
